src/main/java/com/microsoftopentechnologies/windowsazurestorage/WAStorageClient.java

```diff
@@ -335,6 +336,9 @@ public static int upload(Run<?, ?> run, Launcher launcher, TaskListener listener
 			    blob = container.getBlockBlobReference(prefix + srcPrefix);
 			}
 
+			// Set blob properties
+			blobProperties.configure(blob);
```

Comment:
if the user upgrades to the latest plugin and then triggers a build, blobProperties will be null. This is only set to a valid object only after the user configures the job and clicks 'Save'. Have you verified it works ok after upgrade?